File: 0121/main.py
import math
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice: int = 1e9
        price: int = 1e9
        profit: int = 0
        tempProfit: int = 0

        for price in prices:
            if price < minPrice:
                minPrice = price
            else:
                tempProfit = price - minPrice
                if tempProfit > profit:
                    profit = tempProfit
        
        return profit

File: 0121/main.swift
class Solution {
    func maxProfit(_ prices: [Int]) -> Int {
        var minPrice = Int.max
        var profit = 0
        var tempProfit = 0

        for price in prices{
            tempProfit = price - minPrice
            if tempProfit < 0{
                minPrice = price
            } else if tempProfit > 0 && tempProfit > profit{
                profit = tempProfit
            }
        }

        return profit
    }
}